src/language/XMLUtils.js

```diff
@@ -0,0 +1,303 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, $ */
+
+define(function (require, exports, module) {
+    "use strict";
+    
+    // Load dependencies.
+    var TokenUtils  = require("utils/TokenUtils");
+    
+    // Enums of token types.
+    var TOKEN_TAG    = 1,
+        TOKEN_ATTR   = 2,
+        TOKEN_VALUE  = 3;
+    
+    // Regex to find whitespace.
+    var regexWhitespace = /^\s+$/;
+    
+    /**
+     * Returns an object that represents all its params.
+     *
+     * @param {!Token} token CodeMirror token at the current pos
+     * @param {number} tokenType Type of current token
+     * @param {number} offset Offset in current token
+     * @param {Array.<string>} exclusionList List of attributes of a tag or attribute options used by an attribute
+     * @param {string} tagName Name of the current tag
+     * @param {string} attrName Name of the curent attribute
+     * @param {boolean} shouldReplace true if we don't want to append ="" to an attribute
+     * @return {!{token: Token, tokenType: int, offset: int, exclusionList: Array.<string>, tagName: string, attrName: string, shouldReplace: boolean}}
+     */
+    function _createTagInfo(token, tokenType, offset, exclusionList, tagName, attrName, shouldReplace) {
+        return {
+            token: token || null,
+            tokenType: tokenType || null,
+            offset: offset || 0,
+            exclusionList: exclusionList || [],
+            tagName: tagName || "",
+            attrName: attrName || "",
+            shouldReplace: shouldReplace || false
+        };
+    }
+    
+    /**
+     * Return the tagName and a list of attributes used by the tag.
+     *
+     * @param {!Editor} editor An instance of active editor
+     * @param {!{line: number, ch: number}} constPos The position of cursor in the active editor
+     * @return {!{tagName: string, exclusionList: Array.<string>, shouldReplace: boolean}}
+     */
+    function _getTagAttributes(editor, constPos) {
+        var pos, ctx, ctxPrev, ctxNext, ctxTemp, tagName, exclusionList = [], shouldReplace;
+        
+        pos = $.extend({}, constPos);
+        ctx = TokenUtils.getInitialContext(editor._codeMirror, pos);
+        
+        // Stop if the cursor is before =
+        ctxTemp = $.extend(true, {}, ctx);
+        if (ctxTemp.token.type === null && regexWhitespace.test(ctxTemp.token.string)) {
+            if (TokenUtils.moveSkippingWhitespace(TokenUtils.moveNextToken, ctxTemp)) {
+                if (ctxTemp.token.type === null && ctxTemp.token.string === "=") {
+                    return false;
```

Comment:
I understand that it's not a valid place for hints. But since this function returns an `Object` when hints are valid, I think it should return `null` (not `false`) when hints are not valid to be more consistent.